src/main/java/org/opensearch/knn/jni/SIMDEncoding.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.jni;
+
+import org.opensearch.knn.common.KNNConstants;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+public class SIMDEncoding {
+
+    static {
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            try {
+                // Load SIMD library based on support
+                if (PlatformUtils.isAVX512SPRSupportedBySystem()) {
+                    System.loadLibrary(KNNConstants.SIMD_AVX512_SPR_JNI_LIBRARY_NAME);
+                } else if (PlatformUtils.isAVX512SupportedBySystem()) {
+                    System.loadLibrary(KNNConstants.SIMD_AVX512_JNI_LIBRARY_NAME);
+                } else if (PlatformUtils.isAVX2SupportedBySystem()) {
+                    System.loadLibrary(KNNConstants.SIMD_AVX2_JNI_LIBRARY_NAME);
+                } else {
+                    System.loadLibrary(KNNConstants.SIMD_JNI_LIBRARY_NAME);
+                }
+            } catch (UnsatisfiedLinkError e) {
+                throw new RuntimeException("[KNN] Failed to load native SIMD library", e);
```

Comment:
This would also need handling for the fallback logic. 
I think it's better to design for correctness first and then performance. You can add in a log statement here calling out a failure to load these libraries - we do not want to fail all other cases.